packages/ngx-deploy-npm/src/executors/deploy/engine/engine.spec.ts

```diff
@@ -128,4 +136,84 @@ describe('engine', () => {
       expect(setPackage.setPackageVersion).not.toHaveBeenCalled();
     });
   });
+
+  describe('Package Version Check Feature', () => {
+    const mockPackageJson = {
+      name: '@test/package',
+      version: '1.0.0',
+    };
+
+    const versionCheckSetup = ({
+      ...originalSetupOptions
+    }: Parameters<typeof setup>[0] = {}) => {
+      jest
+        .spyOn(fileUtils, 'readFileAsync')
+        .mockImplementation(() =>
+          Promise.resolve(JSON.stringify(mockPackageJson))
+        );
+
+      return setup(originalSetupOptions);
+    };
```

Comment:
suggestion: parametrize `npm view` and `npm publish`.

To cover my suggestion, _thought: cover the unexciting package scenario_ you can do the following to parameterize the behavior of spamAsync depending if it invoking `npm view` or `npm publish`.

```suggestion
    const versionCheckSetup = ({
      npmViewMock = () => Promise.resolve(), // use () => Promise.reject() when you want to induce an error
      npmPublishMock = () => Promise.resolve(),
      ...originalSetupOptions
    }
    : { 
      npmViewMock: () => Promise<void>,
      npmPublishMock: () => Promise<void> 
    } & Omit<Parameters<typeof setup>[0], 'spawnAsyncReturnValue'>) => {
      jest
        .spyOn(fileUtils, 'readFileAsync')
        .mockImplementation(() =>
          Promise.resolve(JSON.stringify(mockPackageJson))
        );
        
      const spawnAsyncReturnValue: ('npm', ['view' | 'publish', ...ortherParams]) =>  = (_, [viewOrPublish]) => {
         return viewOrPublish === 'view' ? npmViewMock: npmPublishMock;
      }

      return setup({
        spawnAsyncReturnValue,
        ...originalSetupOptions
      });
    };
```

###### Note: this code should be treated as pseudo-code.

Reply:
Thanks, I modified a bit the logic. It should be fixed now [with this change](https://github.com/bikecoders/ngx-deploy-npm/compare/d9f3fdb127c7f2fbb667a21dd5c28f82b5d2977c..cc6cfe33f1a9e60cf6e985e0660aa3a347429189)